microsoft-edge/webview2/how-to/winrt-from-js.md

```diff
@@ -0,0 +1,421 @@
+## Step 7. Add the host object in the webview2_sample_uwp project
+
+Next, pass the WinRT object from the native side of the host app to the web side of the host app.  To do this, add an `InitializeWebView2Async` method that calls `AddHostObjectToScript`, as follows:
+
+1. In Solution Explorer, expand the **webview2_sample_uwp** project, and then select **MainPage.xaml.cs**.
+
+1. Below the `MainPage` constructor, add the following `InitializeWebView2Async` method:
+
+   ```csharp
+   private async void InitializeWebView2Async()
+   {
+      await WebView2.EnsureCoreWebView2Async();
+      var dispatchAdapter = new WinRTAdapter.DispatchAdapter();
+      WebView2.CoreWebView2.AddHostObjectToScript("Windows", dispatchAdapter.WrapNamedObject("Windows", dispatchAdapter));
+   }
+   ```
+
+   This method calls `AddHostObjectToScript`.
+
+1. In the `MainPage` constructor, above the `StatusUpdate("Ready");` line, add the following code:
+
+   ```csharp
+   InitializeWebView2Async();
+   ```
+
+1. Right-click the **webview2_sample_uwp** project, and then select **Set as startup project**.
+
+1. Press `Shift+Ctrl+S` to save all files.
+
+1. Press `F5` to run the sample app.  The **WebView2 WinUI 2 UWP Sample** window opens.
+
+   ![The WebView2 WinUI 2 UWP Sample window.](winrt-from-js-images/webview2-winui-2-uwp-sample-app-window.png)
+
+The host app's web-side code (and the DevTools Console) can now call methods and properties of the specified namespaces or classes of the host object.
+
+
+<!-- =============================================== -->
+## Step 8. Call methods and properties on the host object from web-side JavaScript
+
+Next, use the DevTools Console to demonstrate that web-side code can call the included, specified host-side APIs.
+
+1. Click in the main part of the WebView2 sample app window to give it focus, and then press `Ctrl+Shift+I` to open Microsoft Edge DevTools.  Or, right-click the page, and then select **Inspect**.  The Microsoft Edge DevTools window opens.
+
+1. If the Microsoft Edge DevTools window isn't visible, press `Alt+Tab` to display the DevTools window.  If needed, move the DevTools window.
+
+1. In the DevTools Console, paste the following code, and then press `Enter`:
+
+   ```javascript
+   const Windows = chrome.webview.hostObjects.sync.Windows;
+   (new Windows.Globalization.Language("en-US")).displayName;
+   ```
+
+   The Console outputs a language name string, such as `English (United States)`, demonstrating that your app's host-side code can be called from web-side code:
+
+   ![Using the DevTools Console to test calling native-side code from web-side code.](winrt-from-js-images/devtools-console-calling-native-side-code.png)
+
+Congratulations!  You've finished the sample demonstration of calling WinRT code from JavaScript code.
+
+
+<!-- ====================================================================== -->
+## Make AddHostObjectToScript act like Chakra WinRT projection
```

Comment:
@jasonstephen15 are devs landing here expected to have enough context about what Chrakra is or why they'd want WebView2/wv2winrt to behave like it? If not, should a note be added that this is most relevant when migrating an existing codebase from XAML WebView?

Reply:
@Syul968 @david-risney - Are we recommending that all users consider using the code listing below, or only use that code if you are migrating an existing codebase (a WinRT WebView2 host app) from EdgeHTML XAML WinUI 2 WebView to WinUI 2 WebView2?

If we are recommending the below code for all users, then this section is of broader relevance than only when migrating an existing codebase (a WinRT WebView2 host app) from EdgeHTML XAML WinUI 2 WebView to WinUI 2 WebView2.  In that case, the "Chakra" reference might be relevant just to EdgeHTML/WebView users, but the section is relevant to everyone.  Can we downplay "Chakra" and remove it from h2 heading, and come up with a more direct h2 heading?

I need to know What Chrakra is - but more relevantly and directly, need to know why you'd want WebView2 and the wv2winrt tool to behave in a _certain way_ [_directly specify what way_] (which happens to be that of Chakra).

Reply:
You could maybe say this is to make the AddHostObjectToScript JavaScript proxies act more like other JavaScript APIs. And additionally that end developers migrating from JavaScript WinRT projection in JavaScript UWP apps or the EdgeHtml based WebView, may want to use this to better match that behavior.

Reply:
@david-risney - trying eg:
```
If you want to make the `AddHostObjectToScript` JavaScript proxies act more like other JavaScript APIs, use the following approach.

Or, if you are migrating a host app from JavaScript WinRT projection in JavaScript UWP apps, or from the EdgeHTML-based WebView, you may want to use the following approach, to better match that previous behavior.<!-- This section is most relevant if you are migrating an existing codebase (a WinRT WebView2 host app) from EdgeHTML XAML WinUI 2 WebView to WinUI 2 WebView2. -->
```

Reply:
Improved heading & intro re: proxies.  @david-risney